src/main/java/de/tum/cit/aet/artemis/lti/service/DeepLinkingType.java

```
@@ -0,0 +1,17 @@
+package de.tum.cit.aet.artemis.lti.service;
+
+public enum DeepLinkingType {
+
+    EXERCISE, LECTURE, COMPETENCY, LEARNING_PATH, IRIS, UNKNOWN;
```

"Gotta catch 'em all" enum values are a bad practice that usually implie that you don't know what kind of data you are handling. Why is this needed?

---

It was meant more as a failsafe for handling invalid or malformed target links, because I think there is a high possibility of such an error occurring in the future if Artemis URLs change etc.
Unknown might be bad naming tho, I changed it to INVALID and added some specific error handling in the deep linking service for that type.

---

I don't think you understand how `@RequestParam` get parsed. If you really want flexible parsing you need a `Converter`. In any case, you want requests to fail fast and not get parsed and examined until eventually immediately failing. Unless you can recover from an invalid DeepLinkingType this will just create more management overhead in the code.